rs/bitcoin/ckbtc/minter/src/updates/retrieve_btc.rs

```diff
@@ -337,9 +321,15 @@ pub async fn retrieve_btc_with_approval(
     )
     .await?;
 
-    match kyt_check_address(caller, args.address.clone(), args.amount).await {
-        Ok(kyt_result) => {
-            let (_uuid, status, kyt_provider) = kyt_result;
+    let new_kyt_principal = read_state(|s| {
+        s.new_kyt_principal
+            .expect("BUG: upgrade procedure must ensure that the new KYT principal is set")
+            .get()
+            .into()
+    });
+
+    match new_kyt_check_address(new_kyt_principal, args.address.clone()).await {
```

Comment:
Does-it still make sense to run the OFAC check *after* having burned the tokens? I understand that in the case of Chainanalysis one needs to do it that way to ensure that the user pays for the check, but since now `new_kyt_check_address` is basically free we could reverse the order and avoid having to deal with reimbursing the users in case the deposit address is tainted. WDYT?

Reply:
Make sense! How about keeping it still the old way for this PR? I created a ticket as a follow up https://dfinity.atlassian.net/browse/XC-233

Reply:
Yes doing this change in a separate PR/ticket make sense! Thanks for opening the ticket.